tasks/task1/task1b.py

```diff
@@ -0,0 +1,65 @@
+import numpy as np
+import pandas as pd
+import sys
+
+def companyname(filepath):
```

Comment:
Missing documentation. Lots of good declarative language however this could be broken up into a number of smaller more modular functions. You might find this could be simplified.  It looks like there might be elements of this file and Zijun's file that are duplicating code and effort.

Unit tests please!